pyresample/utils.py

```diff
@@ -692,3 +711,411 @@ def convert_def_to_yaml(def_area_file, yaml_area_file):
     with open(yaml_area_file, 'w') as yaml_file:
         for area in areas:
             yaml_file.write(area.create_areas_def())
+
+
+def from_params(area_id, projection, shape=None, upper_left_extent=None, center=None, area_extent=None,
+                resolution=None, radius=None, units=None, rotation=None, **kwargs):
+    """Takes data the user knows and tries to make an area definition from what can be found.
+
+    Parameters
+    ----------
+    area_id : str
+        ID of area
+    projection : dict or str
+        Projection parameters as a proj4_dict or proj4_string
+    description : str, optional
+        Description/name of area. Defaults to area_id
+    proj_id : str, optional
+        ID of projection (being deprecated)
+    units : str, optional
+        Default projection units: meters, radians, or degrees
+    shape : list, optional
+        Number of pixels in the x and y direction (height, width)
+    area_extent : list, optional
+        Area extent as a list (lower_left_x, lower_left_y, upper_right_x, upper_right_y)
+    upper_left_extent : list, optional
+        Upper left corner of upper left pixel (x, y)
+    center : list, optional
+        Center of projection (x, y)
+    resolution : list or float, optional
+        Size of pixels: (dx, dy)
+    radius : list or float, optional
+        Length from the center to the edges of the projection (dx, dy)
+    rotation: float, optional
+        rotation in degrees or radians (negative is cw)
+    nprocs : int, optional
+        Number of processor cores to be used
+    lons : numpy array, optional
+        Grid lons
+    lats : numpy array, optional
+        Grid lats
+    optimize_projection:
+        Whether the projection parameters have to be optimized for a DynamicAreaDefinition.
+
+    Returns
+    -------
+    AreaDefinition or DynamicAreaDefinition : AreaDefinition or DynamicAreaDefinition
+        If shape and area_extent are found, an AreaDefinition object is returned.
+        If only shape or area_extent can be found, a DynamicAreaDefinition object is returned
+
+    Raises
+    ------
+    ValueError:
+        If neither shape nor area_extent could be found
+
+    Notes
+    -----
+    * **units** accepts 'deg', 'degrees', 'rad', 'radians', 'meters', and any parameter from cs2cs
```

Comment:
why not stick to proj style ? m, deg, rad

Reply:
More flexible I guess. I'm ok limiting the amount of unit variations that are allowed. I just thought users might want to specify it different ways so I was ok with this. Thoughts? I suppose limited would be better.

Reply:
then you need to add `metres` :)